downstairs/src/lib.rs

```diff
@@ -2189,8 +2242,14 @@ impl Downstairs {
         m: Message,
         id: ConnectionId,
     ) -> Result<Option<JobId>> {
```

Comment:
I think we need

```rust
        let Some(state) = self.connection_state.get_mut(&id) else {
            warn!(self.log, "got message for disconnected id {id:?}; ignoring");
            return;
        };
```
        
here too?

Reply:
I don't think so – `proc_frame` is only called in `on_message_for`, which does that check itself:

```rust
    async fn on_message_for(&mut self, id: ConnectionId, m: Message) {
        let Some(state) = self.connection_state.get_mut(&id) else {
            warn!(self.log, "got message for disconnected id {id:?}; ignoring");
            return;
        };
        if matches!(state, ConnectionState::Running(..)) {
            match self.proc_frame(m, id).await {
```

I added the panic condition to the `proc_frame` docstring in aeba76b4e25c